bciers/libs/components/src/form/MultiStepBase.tsx

```diff
@@ -86,7 +109,11 @@ const MultiStepBase = ({
     setIsEditMode(true);
   };
 
-  return (
+  return isSuccessfullySubmitted ? (
+    successComponent
+  ) : isSubmitting ? (
+    <div>Submitting...</div>
+  ) : (
```

Comment:
Why are we doing this in here? It feels like we are trying to solve problems that don't exist, the MultiStepBase component is starting to look overly complicated and hard to follow. Now that we have split up the forms into individual pages we can keep logic specific to those pages in those components.

This also dictates that we have to display a different component on submit (we may want to redirect to a page or do something else).

Reply:
> Now that we have split up the forms into individual pages we can keep logic specific to those pages in those components.

There isn't much logic that actually is specific to individual pages, which is why I've moved some things into MultiStepBase. Error handling and `isSubmitting` apply to all form pages. Showing a loading message `<div>Submitting...</div>` while `isSubmitting` is true also seemed general and like it should apply to all steps (previously it was on about half of them, and not the ones I'd expect like the ones with attachments).

Re `successComponent`, previously it was conditionally shown in the last step based on the value of `isSubmitting`. Since `isSubmitting` is now being managed by MultiStepBase, and showing a success message is a general form submission thing, it makes sense to me to have it in MultiStepBase. It could be a redirect to a different page instead, but this was how it was originally set up so I didn't change it.

Reply:
Showing `Submitting...` in place of the entire component is really jarring, and especially noticeable after page 1 with file uploads. I believe we were asked about adding a loading spinner to the `Save and continue` button.

Looking at the `RegistrationSubmissionForm` component it is really clean and simple, I don't see the need to further complicate `MultiStepBase`. We can await the `onSubmit` response like in the example and `setIsSubmitted` and do what we need to in that component if `reponse.ok` or whatever.